<launch>

    <arg name="markerId"        default="5"/>
    <arg name="markerSize"      default="0.01"/>    <!-- in m -->
    <arg name="eye"             default="left"/>
    <arg name="marker_frame"    default="aruco_marker_frame"/>
    <arg name="ref_frame"       default=""/>  <!-- leave empty and the pose will be published wrt param parent_name -->
    <arg name="corner_refinement" default="LINES" /> <!-- NONE, HARRIS, LINES, SUBPIX -->

    <include file="$(find rover_position_control)/launch/rover_pose_control.launch"/>

    <node pkg="aruco_detector" type="triple" name="aruco_simple">    
        <remap from="/camera_info" to="rover/camera/camera_info" />
        <remap from="/image" to="rover/camera/image_raw" />

        <remap from="/camera_info" to="rover/camera/camera_info" />
        <remap from="/image" to="rover/camera/image_raw" />
        <param name="image_is_rectified" value="True"/>
        <param name="marker_size"        value="$(arg markerSize)"/>
        <param name="marker_id"          value="$(arg markerId)"/>
        <param name="reference_frame"    value="$(arg ref_frame)"/>   <!-- frame in which the marker pose will be refered -->
        <param name="camera_frame"       value="stereo_gazebo_$(arg eye)_camera_optical_frame"/>
        <param name="marker_frame"       value="$(arg marker_frame)" />
        <param name="corner_refinement"  value="$(arg corner_refinement)" />
    </node>

    <node pkg="aruco_detector" type="pose" name="aruco_pose" output="screen">
        <rosparam file="$(find aruco_detector)/cfg/aruco.yaml" />
        <!--<rosparam file="$(env HOME)/.ros/camera_info/ocam.yaml"/>-->
        <remap from="odom" to="rover/odom"/>
        <remap from="pose_convert" to="pose"/>
        <!--<remap from="pose2" to="detector/pose2"/>
        <remap from="pose3" to="detector/pose3"/>
        <remap from="flags" to="detector/flags"/>-->
    </node>

</launch>
